docs/curl/provenance.md

```diff
@@ -0,0 +1,829 @@
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      },
+      {
+        "key": "key3",
+        "value": "dGhpcyBpcyBhIHRoaXJkIHZhbHVl",
+        "metadata": {
+          "version": {
+            "block_num": 5
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      },
+      {
+        "key": "key2",
+        "value": "dGhpcyBpcyBhIHNlY29uZCB2YWx1ZSB1cGRhdGVk",
+        "metadata": {
+          "version": {
+            "block_num": 6
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      },
+      {
+        "key": "key1",
+        "value": "dGhpcyBpcyBhIGZpcnN0IHZhbHVlIHVwZGF0ZWQ=",
+        "metadata": {
+          "version": {
+            "block_num": 6
+          },
+          "access_control": {
+            "read_users": {
+              "alice": true
+            },
+            "read_write_users": {
+              "alice": true
+            }
+          }
+        }
+      }
+    ]
+  },
+  "signature": "MEUCIQDaMpeJhgVlFOS0AL7AOd2wviQ6c9R/PrOH51grBifRlgIgL4hLscpLE+JNAryd0it+WkuFmiQyUHbg25/Z+T9ty3A="
+}
+```
+
+As you can see, user `alice` wrote 3 times to key `key1`, in blocks 4, 5 and 6, two times to `key2`, in blocks 5 and 6 and one time to `key3`, in block 5
+
+### Key access queries
+Another important view on data history is to know what user accessed specific key over time. Two next queries provide us with this information.
+
+### Query for key readers
+To query for users who ever read specific key, we use `/provenance/data/readers/{dbname}/{key}` GET query.
+
+**Sign json marshalled query**
+```sh
+bin/signer -data '{"user_id":"alice","db_name":"db2","key":"key1"}' -privatekey=deployment/sample/crypto/alice/alice.key
+```
+
+**Signature**
+```
+MEUCIQDAO/58FDUjaKx+aAN6D5GCsp87hQ0CePeBwxa3FKzPpQIgASJwGGk8rHtZPgKUEMJ7U4H5L8ttQwnXq7XrSCcL/B4=
+```
+
+**Submit query**
+```sh
+curl \
+     -H "Content-Type: application/json" \
+     -H "UserID: alice" \
+     -H "Signature: MEUCIQDAO/58FDUjaKx+aAN6D5GCsp87hQ0CePeBwxa3FKzPpQIgASJwGGk8rHtZPgKUEMJ7U4H5L8ttQwnXq7XrSCcL/B4=" \
+     -X GET http://127.0.0.1:6001/provenance/data/readers/db2/key1 | jq .
+```
+
+**Output**
+```json
+{
+  "response": {
+    "header": {
+      "node_id": "bdb-node-1"
+    },
+    "read_by": {
+      "alice": 1
+    }
+  },
+  "signature": "MEQCIDI6cs21gAUfE/4l+/ha9ih9wPQ8YnnogpzMRt5XrDR9AiAfOsKf9FhUpeYPmMCvM9gIF0YQO6IFtdZUC8khOGTIcA=="
+}
+```
+
+User `alice`read the `key1` value one time.
+
+### Query for key writers
+To query for users who ever wrote to specific key, we use `/provenance/data/writers/{dbname}/{key}` GET query.
```

Comment:
To query for all the users that ever wrote to a specific key, we use `/provenance/data/writers/{dbname}/{key}` GET query. This will return ... i.e.
user : num_times_written

Reply:
done